addons/hr_work_entry_holidays/models/hr_leave.py

```diff
@@ -82,8 +82,9 @@ def _cancel_work_entry_conflict(self):
                 # Intervals are outside, but associated records are overlapping.
                 outside_intervals = conflicts_intervals - leave_intervals
 
-                overlappping |= self.env['hr.work.entry']._from_intervals(outside_intervals)
+                overlappping |= self.env['hr.work.entry']._from_intervals(outside_intervals).filtered(lambda entry: entry.state != 'validated')
                 included |= previous_employee_work_entries - overlappping
+                included = included.filtered(lambda entry: entry.state != 'validated')
             overlappping.write({'leave_id': False})
             included.write({'active': False})
```

Comment:
to avoid going through the whole array multiple times while being in the loop, maybe you could filter before the write:
```suggestion
                overlappping |= self.env['hr.work.entry']._from_intervals(outside_intervals)
                included |= previous_employee_work_entries - overlappping
            overlappping.filtered(lambda entry: entry.state != 'validated').write({'leave_id': False})
            included.filtered(lambda entry: entry.state != 'validated').write({'active': False})
```